app/lib/blazer_admin_constraint.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class BlazerAdminConstraint
+  def matches?(request)
+    signin_user = UserSession.load_from_session(request.session)
+    return false if signin_user.blank?
+
+    admin_user = User.admins.kept.find_by(email: signin_user.email)
+    admin_user.present? && authorised_for_blazer?(admin_user)
+  end
+
+  private
+
+  def authorised_for_blazer?(admin_user)
+    allowed_ids.blank? || allowed_ids.split(',').include?(admin_user.id.to_s)
+  end
+
+  def allowed_ids
+    ENV.fetch('BLAZER_ALLOWED_IDS', nil)
```

Comment:
Why have we chosen to store this in an environment variable? 

I think I would prefer it to be a column on the user model. Then we (as devs) can just use the rails console to add / remove blazer access, rather than the rather cumbersome step of updating environment variables and redeploying the app. If we keep the logic on the user, we can essentially delete lines 12 - 19 and change lines 8-9 to:
`User.admins_with_blazer_access.find_by(email: singin_user.email).present?` 
(where admins_with_blazer_access) is a scope on the User model)

The implementation here also means our default position is that all admin users have blazer access (if there is no env var). Is that what we want, or would we prefer that blazer access is always restricted to particular users? 

I'm not going to block this over it, but I'm curious as to why we have chosen to implement it this way. In general, it feels like making this dependent on an env var spreads the logic about who is / isn't a blazer user to more places than it needs to be. Making it harder to debug and read in the future.

Reply:
Agreed to go with this approach offline.